### ISSUE TITLE
SIGSEGV when calling vkDestroyImageView while validation layers are enabled

### ISSUE THREAD

**Issue by Scoopta** [KhronosGroup/Vulkan-LoaderAndValidationLayers#1331]
I'm getting this really weird issue where I get a SIGSEGV when I call vkDestoryImageView while having VK_LAYER_LUNARG_standard_validation and VK_EXT_DEBUG_REPORT_EXTENSION enabled. Without the validation layer everything works fine but once I enable it I get a seg fault here:
`[libVkLayer_core_validation.so+0x153749]  core_validation::DestroyImageView(VkDevice_T*, VkImageView_T*, VkAllocationCallbacks const*)+0xf9`
If I comment out vkDestroyImageView the validation layer works as intended and tells me that I need to do destroy all child objects before destroying the device. The other really really odd thing is I have a total of 3 views in an array. If I only destroy the view at array index 0 it works fine. The validation layer notifies me that I need to destroy the other 2 and all is well. However attempting to destroy the views in index 1 or 2 causes the seg fault even if it's the only view I attempt to destroy. If I disable the validation layer then all three views are destroyed and everything works as intended without any problems or crashes.

**Comment by tobine** [comment]
I checked this code and from inspection I have no idea what may be wrong. @Scoopta do you have a repro case you can share?
I don't doubt that there could be an issue here, but vkDestroyImageView shares a common pattern with a number of destroy functions and I haven't heard of any similar issues.

**Comment by Scoopta** [comment]
I can post my code but there's a few things. First, this is my first time working with Vulkan and I was more or less following [Vulkan Tutorial](https://vulkan-tutorial.com) although it's not identical. The code is really messy and probably hard to follow because it's my first time working with Vulkan. The second thing is this was done in Java using LWJGL's Vulkan binding. Here's the code. I don't have an existing repo setup for it on bitbucket so I just threw it on pastebin: [http://pastebin.com/YVnNKbfG](http://pastebin.com/YVnNKbfG)

**Comment by mark-lunarg** [comment]
@Scoopta, another thing that could be useful is to share an API dump of your app up to and including the validation error.  You can load the api_dump layer in addition to standard_validation and we should be able to diagnose the issue that way if getting the repro case going takes too long.

**Comment by Scoopta** [comment]
Alright I'll do that.

**Comment by Scoopta** [comment]
@mark-lunarg alright I got it figured out. It turns out the other 2 pointers in the array were 0. I'm not entirely sure why I was only getting the SIGSEGV with the validation layers enabled. I would think that would be triggered regardless but apparently not.
